app/src/main/java/cm/aptoide/pt/discovery/VideoViewHolder.java

```diff
@@ -0,0 +1,115 @@
+import android.widget.TextView;
+import android.widget.VideoView;
+import cm.aptoide.pt.R;
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.load.DataSource;
+import com.bumptech.glide.load.engine.DiskCacheStrategy;
+import com.bumptech.glide.load.engine.GlideException;
+import com.bumptech.glide.request.RequestListener;
+import com.bumptech.glide.request.target.Target;
+
+public class VideoViewHolder extends RecyclerView.ViewHolder {
+
+  private static final String TAG = "VIDEOVIEWHOLDER";
+  private CoordinatorLayout appInfoBackground;
+  private Context context;
+
+  private VideoView videoContent;
+  private TextView appTitle;
+  private TextView appScore;
+  private ImageView appIcon;
+  private ImageView ratingStar;
+  private Button installButton;
+
+  private ImageView testImageView;
+
+  private Palette.Swatch paletteSwatch;
+
+  private Bitmap bitmap;
+
+  public VideoViewHolder(View itemView) {
+    super(itemView);
+    videoContent = (VideoView) itemView.findViewById(R.id.app_video);
+    appTitle = (TextView) itemView.findViewById(R.id.app_title);
+    appScore = (TextView) itemView.findViewById(R.id.app_score);
+    appIcon = (ImageView) itemView.findViewById(R.id.discovery_app_icon);
+    ratingStar = (ImageView) itemView.findViewById(R.id.rating_star);
+    appInfoBackground = (CoordinatorLayout) itemView.findViewById(R.id.app_info_bar);
+    installButton = (Button) itemView.findViewById(R.id.install_button);
+
+    this.context = itemView.getContext();
+  }
+
+  public void setContent(Video video) {
+    videoContent.setVideoPath(video.getVideoUrl());
+    videoContent.requestFocus();
+    videoContent.start();
+  }
+
+  public void setAppName(Video video) {
+    appTitle.setText(video.getVideoDescription());
+  }
+
+  public void setAppScore(Video video) {
+    appScore.setText(Double.toString(video.getRating()));
+  }
+
+  public void setAppIcon(Video video) {
+    Glide.with(this.context)
+        .asBitmap()
+        .load(video.getImageUrl())
+        .into(appIcon);
+  }
+
+  public void setAppInfoBackgroundColour(Video video) {
+
+    GlideApp.with(this.context)
+        .asBitmap()
+        .load(video.getImageUrl())
+        .disallowHardwareConfig()
+        .diskCacheStrategy(DiskCacheStrategy.ALL)
+        .placeholder(R.mipmap.ic_launcher)
+        .listener(new RequestListener<Bitmap>() {
+
+          @Override public boolean onLoadFailed(@Nullable GlideException e, Object model,
+              Target<Bitmap> target, boolean isFirstResource) {
+            return false;
+          }
+
+          @Override
+          public boolean onResourceReady(Bitmap resource, Object model, Target<Bitmap> target,
+              DataSource dataSource, boolean isFirstResource) {
+
+            if (resource != null) {
+              Palette.Swatch p = Palette.from(resource)
+                  .generate()
+                  .getVibrantSwatch();
+              appInfoBackground.setBackgroundColor(p.getRgb());
```

Comment:
according to the getVibrantSwatch() -> "Returns the most vibrant swatch in the palette. Might be null."
we might get a npe here.

Reply:
fixed